docs/src/pages/components/switchfield/SwitchPropControls.tsx

```diff
@@ -0,0 +1,158 @@
+  ) => void;
+}
+
+interface SwitchPropControlsInterface {
+  (props: SwitchPropControlsProps): JSX.Element;
+}
+
+export const SwitchPropControls: SwitchPropControlsInterface = ({
+  thumbColor,
+  setThumbColor,
+  trackColor,
+  setTrackColor,
+  trackCheckedColor,
+  setTrackCheckedColor,
+  isDisabled,
+  setIsDisabled,
+  size,
+  setSize,
+  label,
+  setLabel,
+  labelPosition,
+  setLabelPosition,
+  isLabelHidden,
+  setIsLabelHidden,
+  isChecked,
+  setIsChecked,
+}) => {
+  return (
+    <Flex direction="column">
+      <TextField
+        placeholder="Set thumb color"
+        name="thumbColor"
+        value={thumbColor}
+        onChange={(event: any) => {
+          setThumbColor(event.target.value);
+        }}
+        label="thumbColor"
+      />
+      <TextField
+        placeholder="Set track color"
+        name="trackColor"
+        value={trackColor}
+        onChange={(event: any) => {
+          setTrackColor(event.target.value);
+        }}
+        label="trackColor"
+      />
+      <TextField
+        placeholder="Set checked track color"
+        name="trackCheckedColor"
+        value={trackCheckedColor}
+        onChange={(event: any) => {
+          setTrackCheckedColor(event.target.value);
+        }}
+        label="trackCheckedColor"
+      />
+      <SelectField
+        name="size"
+        value={size}
+        placeholder="default"
+        onChange={(event) =>
+          setSize(event.target.value as SwitchFieldProps['size'])
+        }
+        label="size"
+      >
+        <option value="small">small</option>
+        <option value="large">large</option>
+      </SelectField>
+      <TextField
+        placeholder="Set switchfield label"
+        name="label"
+        value={label as string}
+        onChange={(event: any) => {
+          setLabel(event.target.value);
+        }}
+        label="label"
+      />
+      <SelectField
+        name="labelPosition"
+        value={labelPosition}
+        onChange={(event) =>
+          setLabelPosition(
+            event.target.value as SwitchFieldProps['labelPosition']
+          )
+        }
+        label="labelPosition"
+      >
+        <option value="start">start</option>
+        <option value="end">end</option>
+        <option value="top">top</option>
+        <option value="bottom">bottom</option>
+      </SelectField>
+      <SwitchField
+        name="isLabelHidden"
+        onChange={() => {
+          setIsLabelHidden(!isLabelHidden);
+        }}
+        label="isLabelHidden"
+      />
+      <SwitchField
+        name="isDisabled"
+        onChange={() => {
+          setIsDisabled(!isDisabled);
+        }}
+        label="isDisabled"
+      />
+      <Flex>
+        <SwitchField
+          name="isChecked"
+          onChange={() => {
+            setIsChecked(!isChecked);
+          }}
+          label="isChecked"
+        />
+        <Button
+          variation="link"
+          onClick={() => setIsChecked(undefined)}
```

Comment:
This button doesn't seem to do anything when I click it. It should control whether it's switched on or off, correct?

Reply:
That button is used to clear the controlled state set by `isChecked`. When it works as a controlled component, you cannot change its on/off state by simply click on it